kubetest2-gce/deployer/deployer.go

```diff
@@ -81,6 +81,27 @@ type deployer struct {
 	CreateCustomNetwork         bool   `desc:"Sets the environment variable CREATE_CUSTOM_NETWORK=true during deployment."`
 }
 
+// pseudoUniqueSubstring returns a substring of a UUID
+// that can be reasonably used in resource names
+// where length is constrained
+// e.g https://cloud.google.com/compute/docs/naming-resources
+// but still retain as much uniqueness as possible
+// also easily lets us tie it back to a run
+func pseudoUniqueSubstring(uuid string) string {
+	// both KUBETEST2_RUN_ID and PROW_JOB_ID uuids are generated
+	// following RFC 4122 https://tools.ietf.org/html/rfc4122
+	// e.g. 09a2565a-7ac6-11eb-a603-2218f636630c
+	// extract the first 13 characters (09a2565a-7ac6) as they are the ones that depend on
```

Comment:
if we're constrained to 12/36 characters, perhaps we should skip the fixed `-`?

Reply:
we also probably don't want purely timestamp bits if we can avoid it, else concurrent runs can conflict ...?

Reply:
I mean currently it's a fixed name for all so it's no worse